hydroflow/src/compiled/pull/symmetric_hash_join.rs

```diff
@@ -22,29 +23,37 @@ where
     ValBuild: Clone + Eq,
     ValProbe: Clone,
 {
-    fn pop_buffer(&mut self) -> Option<(Key, ValProbe, ValBuild)> {
-        let (k, v, idx) = self.current_matches.as_mut()?;
-        let vec = &self.table[k];
-        let result = (k.clone(), v.clone(), vec[*idx].clone());
-        *idx += 1;
-        if vec.len() <= *idx {
-            self.current_matches = None;
-        }
-        Some(result)
-    }
-
     fn build(&mut self, k: Key, v: &ValBuild) -> bool {
-        let vec = self.table.entry(k).or_insert_with(Vec::new);
-        if !vec.contains(v) {
-            vec.push(v.clone());
-            return true;
-        }
+        let entry = self.table.entry(k);
+
+        match entry {
+            Entry::Occupied(mut e) => {
+                let vec = e.get_mut();
+
+                if !vec.contains(v) {
+                    vec.push(v.clone());
+                    return true;
+                }
+            }
+            Entry::Vacant(e) => {
+                e.insert(vec![v.clone()]);
+                return true;
+            }
+        };
+
         false
     }
 
-    fn probe(&mut self, k: Key, v: ValProbe) {
-        if self.table.contains_key(&k) {
-            self.current_matches = Some((k, v, 0));
+    fn probe(&mut self, k: &Key, v: &ValProbe) {
```

Comment:
As far as any other optimizations, maybe we could handle the single match case directly without buffering by returning an `Option<match>` here. But otherwise LGTM